desktop/libs/notebook/src/notebook/api.py

```diff
@@ -405,7 +405,7 @@ def cancel_statement(request):
   response = {'status': -1}
 
   notebook = json.loads(request.POST.get('notebook', '{}'))
-  snippet = None
+  snippet = json.loads(request.POST.get('snippet', None))
   operation_id = request.POST.get('operationId') or notebook['uuid']
 
   snippet = _get_snippet(request.user, notebook, snippet, operation_id)
```

Comment:
Quick question: We get the snippet here from the notebook, so does it mean notebook doesn't have snippet for close_statement call?